core/web/schema/type/error.graphql

```diff
@@ -2,6 +2,7 @@ enum ErrorCode {
 	NOT_FOUND
 	INVALID_INPUT
 	UNPROCESSABLE
+	STATUS_CONFLICT
```

Comment:
Minor: do we need a new error code for this case or can we just use "unprocessable"? Another thing to note is we probably want to move away from error codes since on the frontend we will simply match on the full error type in the union.

Reply:
I would call this an InvalidInput and use the `InputErrors` type as you are performing a check on one of the input fields `oldPassword`.

On a side note.
I've been thinking a bit more about the errors and I'm actually leaning towards leaving it as is and returning top level GQL errors as the default error mechanism. Thinking about it from a UI perspective, it will handle GQLError error messages (without an extension code) in the same way each time (probably by displaying an alert). Since the GQLError only has a top level message this is easy for them to handle as a default case. When we want to expose more granular error message, we could move them up into the payload.

I know this breaks the idea of leaving the GQLErrors only for Server level errors, but I think this gives us more flexibility in the future to adjust if we need to, and reduces the amount of code we would need to write to add Errors to every Payload union type.

Thoughts?

Reply:
I agree with you @jkongie, it feels like we would need the flexibility moving forward. And as you just stated, the amount of workload gets reduced each time we do something new.

What do you think @makramkd ?

Reply:
Sounds good to me. Definitely don't want to put more work on our side and on the frontend to handle an increasingly large set of error types. 👍